src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -681,63 +683,46 @@ public int restandardizeParentStructures(List<Long> parentIds)
 			}
 			// Update the percentage.
 			previousPercent = percent;
+
+			remainingCount = StandardizationDryRunCompound.getReadyStandardizationChangesCount();
 		}
 		return(p);
 	}
 	
 	@Transactional(propagation = Propagation.REQUIRES_NEW)
-	public int restandardizeParentStructuresBatch(List<Long> pIdGroup) throws CmpdRegMolFormatException, StandardizerException, IOException {
-		// Create standardization hashmap
-		HashMap<String, String> parentIdsToStructures = new HashMap<String, String>();
-		HashMap<Long, Parent> parents = new HashMap<Long, Parent>();
-		HashMap<Long, String> parentAsDrawnMap = Lot.getOriginallyDrawnAsStructuresByParentIds(pIdGroup);
-		Parent parent;
-		String standardizedMol;
-
-		for (Long parentId : pIdGroup) {
-			parent = Parent.findParent(parentId);
-			parents.put(parentId, parent);
-
-			// Get the as drawn structure
-			String asDrawnStruct = parentAsDrawnMap.get(parent.getId());
-			if (asDrawnStruct == null) {
-				logger.warn("Parent " + parentId + " has no as drawn structure");
-				parentIdsToStructures.put(parentId.toString(parentId), parent.getMolStructure());
-			} else {
-				parentIdsToStructures.put(parentId.toString(parentId), asDrawnStruct);
-			}
-
+	public int restandardizeParentsOfStandardizationDryRunCompoundsBatch(int batchSize) throws CmpdRegMolFormatException, StandardizerException, IOException {
+
+		// Create a hashmap we can use to store the cmpdreg molecules for each cdId in the standardizationDryRunCompounds
+		HashMap<String, Integer> cdIdMap = new HashMap<String, Integer>();
+		List<StandardizationDryRunCompound> standardizationDryRunCompounds = StandardizationDryRunCompound.findReadyStandardizationChanges().setMaxResults(batchSize).getResultList();
+		for(StandardizationDryRunCompound c : standardizationDryRunCompounds) {
+			// For each standardization compound we want to make a hash of it id and cdId
+			cdIdMap.put(String.valueOf(c.getId()), c.getCdId());
 		}
+		// Get the cmpd reg molecules from the dry run standardization
+		HashMap<String, CmpdRegMolecule> cmpdRegMolecules = chemStructureService.getCmpdRegMolecules(cdIdMap, StructureType.STANDARDIZATION_DRY_RUN);
 
-		// Do standardization
-		logger.info("Starting standardization of " + parentIdsToStructures.size() + " compounds");
-		// Start timer
-		long standardizationStart = new Date().getTime();
-		HashMap<String, CmpdRegMolecule> standardizationResults = chemStructureService
-				.standardizeStructures(parentIdsToStructures);
```

Comment:
This is the good part of removing the call to standardize the parents AGAIN unnecessarily and instead just fetch the cmpdRegmolecules (703)